lua/telescope/utils.lua

```diff
@@ -284,4 +284,244 @@ function utils.strcharpart(str, nchar, charlen)
+    else
+        if newcol < wincol then
+            return newcol + 1
+        elseif newcol == wincol then
+            return wincol
+        else
+            return utils.edge_wincol({pos[1], pos[2] - 1}, wincol, false)
+        end
+    end
+end
+
+-- Greedily get closest byte column left {true} or right {left=false} from passed wincol
+-- Once edge wincol for blockwise selection are found, this function finds
+-- closest or match byte column in terms of starting pos and direction
+-- Returns byte column zero indexed
+-- Note: greedy resolution is performed as recursion would otherwise require difficult/opaque edge cases to catch
+-- @param wincol integer wincol to get closest or equal byte column for
+-- @param pos table {row and column} {1, 0}-index of position to resolve
+-- @param line_len integer length of line in byte columns, zero-indexed (lua string length - 1)
+-- @param left boolean approximate or match wincol from left (true, ~max) or from right (false, ~min)
+function utils.greedy_wincol_byte(wincol, pos, line_len, left)
+    left = left or false
+    local max_col = utils.get_wincol({pos[1], 2^31-1})
+
+    if left then
+        if wincol > max_col then
+            return line_len
+        end
+        for i=0,line_len do
+            local wcol = utils.get_wincol({pos[1], i})
+            if wcol == wincol then
+                return i
+            end
+            if wcol > wincol then
+                return i - 1
+            end
+        end
+     else
+        if wincol > max_col then
+            return nil
+        end
+        for i=line_len,0,-1 do
+            local wcol = utils.get_wincol({pos[1], i})
+            if wcol <= wincol then
+                return i
+            end
+        end
+
+    end
+end
+
+-- Recursively resolve beginning or ending byte column for multi-width characters
+--  Example: multi-width Japanese Zenkaku, see change in columns from left to right char
+--  全角
+-- For 全 above and offset=1, resolve_bytecol returns 7
+-- vim.str_byteindex is similar though can only return the right-hand border
+-- Note: byte columns are 1-indexed for simplified interop with lua string:sub
+-- @param line string: buffer line
+-- @param byte_col integer: initial byte column
+-- @param offset integer: resolver char border towards char beginning (-1) or char end (+1)
+-- return byte_col integer: byte column of char beginning or end
+function utils.resolve_bytecol(line, byte_col, offset)
+    local utf_start, _ = vim.str_utfindex(line, math.min(byte_col, #line))
+    local utf_start_offset, _ = vim.str_utfindex(line, math.min(byte_col + offset, #line))
+    if utf_start == utf_start_offset and byte_col + offset <= #line and byte_col + offset >= 0 then
+        return utils.resolve_bytecol(line, byte_col + offset, offset)
+    else
+        return byte_col
+    end
+end
+
+function utils.get_visual_selection(delimiter, trim, exit_visual_mode)
+    delimiter = delimiter or ' '
+    trim = trim or true
```

Comment:
use vim.F.if_nil

Reply:
Thanks for your comments! The `x` or true are a bit depressing :) saw a similar pattern somewhere once and copied that, though thinking about it after your comment.. it's a bit painfully obvious I'm afraid haha

Reply:
_everybody_ gets bitten by that at some point :)

Reply:
:laughing: actually conni and tj told me about it, once I went crazy debug and that was the reason :laughing: . 

basically vim.F.if_nil is the following

```lua
x = x == nil and true or x

function F.if_nil(a, b)
  return a == nil and b or a
end

```